types.d.ts

```diff
@@ -1042,6 +1017,31 @@ declare var mage: Mage;
 
 declare namespace mage {
     namespace archivist {
+        /**
+         * The media type defines how the data will be deserialized when read from a topic, and
+         * how it will be serialized before it is written to a topic.
+         */
+        type ArchivistMediaType = 'application/json' | 'application/octet-stream' | 'application/x-tome' | 'text/plain' | string;
+
+        /**
+         * Defines to what type of data structure a topic instance will be deserialized into. It is also used
+         * in some cases to specify to a serialization method what is the data type of the data we are feeding in.
+         */
+        type ArchivistEncoding = 'utf8' | 'buffer' | 'live';
+
+        /**
+         * Callback functions
+         */
+        type ArchivistExistsCallback = (error: Error|null, doesValueExist: boolean) => void;
+
+        type ArchivistGetCallback<T> = (error: Error|null, value: T) => void;
+
+        type ArchivistMGetCallback<T> = (error: Error|null, value: T[]) => void;
```

Comment:
This can also be a key/value object, not just an array. Depends on the input to mget basically.